### ISSUE TITLE
LOSS?

### ISSUE THREAD

**Issue by a897456** [facebookresearch/AudioDec#19]
![image](https://github.com/facebookresearch/AudioDec/assets/28752526/76c825f4-335e-4c10-86fd-ebb79f13b751)
# data: LibriTTS   tpye: Autoencoder
Hi, @bigpon 
Excuse me, is it normal for generator loss to increase all the time?
What the correct loss for the generator and discriminator should be ?
THS

**Comment by a897456** [comment]
[train]:  41%|████      | 204200/500000 [18:39:38<23:39:20,  3.47it/s]2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/train/ppl_0 = 1.0007.
2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/train/ppl_1 = 1.0007.
2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/train/ppl_2 = 1.0007.
2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/train/ppl_3 = 1.0074.
2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/train/ppl_4 = 1.0074.
2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/train/ppl_5 = 1.0074.
2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/train/ppl_6 = 1.0155.
2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/train/ppl_7 = 1.0155.
2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/train/vqloss = 9.1469.
2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/mel_loss = 66.0485.
2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/feature_matching_loss = 37.8994.
2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/adversarial_loss = 82.7789.
2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/generator_loss = 157.9743.
2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/real_loss = 0.5400.
2024-04-08 13:55:13,341 (trainerGAN:333) INFO: (Steps: 204200) train/fake_loss = 0.2713.
2024-04-08 13:55:13,342 (trainerGAN:333) INFO: (Steps: 204200) train/discriminator_loss = 0.8113.
[train]:  41%|████      | 204300/500000 [18:40:07<23:36:33,  3.48it/s]2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/train/ppl_0 = 1.0004.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/train/ppl_1 = 1.0004.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/train/ppl_2 = 1.0004.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/train/ppl_3 = 1.0052.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/train/ppl_4 = 1.0052.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/train/ppl_5 = 1.0052.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/train/ppl_6 = 1.0104.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/train/ppl_7 = 1.0104.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/train/vqloss = 9.1468.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/mel_loss = 65.4741.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/feature_matching_loss = 36.5214.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/adversarial_loss = 80.1363.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/generator_loss = 154.7572.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/real_loss = 0.4934.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/fake_loss = 0.2463.
2024-04-08 13:55:42,413 (trainerGAN:333) INFO: (Steps: 204300) train/discriminator_loss = 0.7397.

**Comment by a897456** [comment]
#  Hi@bigpon 
I changed the dataset from LibriTTS to LibriSpeech, everything become better than the first test like the loss as follows: 
![image](https://github.com/facebookresearch/AudioDec/assets/28752526/bd43db34-3d3a-4920-be78-a962ed4722af)
# but when the 400k step is completed, the voice sounds like an electric current. Do you know the reason? You can listen some output files in here:
https://drive.google.com/drive/folders/1p2K09Am-Paz4I-H39uXYj1pOzFiTBjeL?usp=drive_link

**Comment by bigpon** [comment]
I assume that you don't use the GAN loss, so the reconstructed audio suffers the buzzy noise.
GAN training is essential for AudioDec.

For the default settings of the LibriTTS, we warm up the model using only metric losses for the first 500K iters, and continuously train it with both metric and GAN losses for the following 500K iters.

Please check the paper and script details.

**Comment by a897456** [comment]
> I assume that you don't use the GAN loss, so the reconstructed audio suffers the buzzy noise. GAN training is essential for AudioDec.

Yes, I initially thought the 500k step training was the SoundStream training, I heard your pre-trained model output, the same effect.

> For the default settings of the LibriTTS, we warm up the model using only metric losses for the first 500K iters, and continuously train it with both metric and GAN losses for the following 500K iters.

I added the wav discriminator and stft discriminator, ran 1000k of soundstream training, and found that the results were better than symAD(msd and mpd discriminator replace wav and stft discriminator), both of which used the librispeech dataset, so I donot know the reason, I think your article is right that msd and mpd discriminator is better than wav and stft discriminator, but the actual result is not so. can you point out the reason to me?

**Comment by bigpon** [comment]
In our re-implemented soundstream, we used MSDs + single-resolution STFTD.
In our paper, we only claim that using MSDs + MPDs is better than using MSDs + single-resolution STFTD.

For our internal experiments, the more types of discriminators are adopted  the better the performance is.
However, using MPDs + multi-resolution STFTDs might be good enough, which is also matched to other related works' results.

**Comment by a897456** [comment]
> In our re-implemented soundstream, we used MSDs + single-resolution STFTD. In our paper, we only claim that using MSDs + MPDs is better than using MSDs + single-resolution STFTD.
> 
> For our internal experiments, the more types of discriminators are adopted the better the performance is. However, using MPDs + multi-resolution STFTDs might be good enough, which is also matched to other related works' results.

so nice work, THS

**Comment by a897456** [comment]
> For our internal experiments, the more types of discriminators are adopted the better the performance is. However, using MPDs + multi-resolution STFTDs might be good enough, which is also matched to other related works' results.

Hi @bigpon 
Excuse me again, I saw that UnivNet proposed MSRD(multi-resolution spectrogram discriminator). Is it the same thing with the multi-resolution STFTDs you mentioned?

**Comment by bigpon** [comment]
Hi, 
Yes, I adopt it from [SiFiGAN repo](https://github.com/chomeyama/SiFiGAN/blob/main/sifigan/models/discriminator.py).
